src/v0/destinations/fb_custom_audience/recordTransform.js

```diff
@@ -235,6 +220,75 @@ async function processRecordInputs(groupedRecordInputs) {
   return finalResponse;
 }
 
+async function processRecordInputsV1(groupedRecordInputs) {
+  const { destination } = groupedRecordInputs[0];
+  const { message } = groupedRecordInputs[0];
+  const { isHashRequired, disableFormat, type, subType, isRaw, audienceId, userSchema } =
+    destination.Config;
+
+  // mapped to destination
+  const mappedToDestination = get(message, MappedToDestinationKey);
+  let operationAudienceId = audienceId;
+  if (mappedToDestination) {
+    const { objectType } = getDestinationExternalIDInfoForRetl(message, 'FB_CUSTOM_AUDIENCE');
+    operationAudienceId = objectType;
+  }
+
+  // user schema override
+  let updatedUserSchema = userSchema;
+  if (mappedToDestination) {
+    updatedUserSchema = getSchemaForEventMappedToDest(message);
+  }
+
+  return preparePayload(groupedRecordInputs, {
+    audienceId: operationAudienceId,
+    userSchema: updatedUserSchema,
+    isRaw,
+    type,
+    subType,
+    isHashRequired,
+    disableFormat,
+  });
+}
+
+const processRecordInputsV2 = async (groupedRecordInputs) => {
+  const { connection, message } = groupedRecordInputs[0];
```

Comment:
Does transformer get connection in each event? I remember you mentioned this too we can look at in future for potential optimization.

Reply:
yes currently this is like this but @ktgowtham was planning some optimization

Reply:
Yeah. We should do that optimization. @RanjeetMishra , this has been like that. we were getting destConfig, dest def for each event, which are huge objects. This needs to be a combined integrations, DM, server effort. cc: @krishna2020